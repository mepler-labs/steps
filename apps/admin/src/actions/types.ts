export type DispatchFn = (dispatch?: any, getState?: any) => any;
