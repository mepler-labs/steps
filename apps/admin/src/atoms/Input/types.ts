export type InputType = 'text' | 'textarea';
