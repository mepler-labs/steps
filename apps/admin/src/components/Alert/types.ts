export enum AlertLevel {
  Info = 'info',
  Warning = 'warning',
  Error = 'error',
}

export interface Alert {
  level: AlertLevel;
  message: string;
  id: string;
}
