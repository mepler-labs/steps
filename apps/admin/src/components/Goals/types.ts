export type Goal = {
  text: string;
};
