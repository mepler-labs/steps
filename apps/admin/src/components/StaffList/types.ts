export interface PendingInvite {
  pendingInvite: true;
  email: string;
}

export type Staff<T> = {
  pendingInvite: false;
} & T;

export interface User {
  name: string;
  permissionLevel: PermissionLevel;
  email: string;
}

export enum PermissionLevel {
  Administrator = 'administrator',
  Coach = 'coach',
}

export type StaffMember = PendingInvite | Staff<User>;
