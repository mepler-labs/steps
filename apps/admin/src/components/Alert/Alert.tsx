import styled from 'styled-components';
import * as React from 'react';
import { AlertLevel } from './types';
import { red, white, yellow, blue, green } from 'styles/colors';
import { Flex } from 'grid-styled';

export interface Props {
  level: AlertLevel;
  onClose(event): void;
}

export const Alert = ({ level, children, onClose }) => {
  return (
    <StyledAlert level={level}>
      <Flex justifyContent="space-between" alignItems="center">
        {children}
        <CloseIcon className="material-icons" onClick={onClose}>
          clear
        </CloseIcon>
      </Flex>
    </StyledAlert>
  );
};

const alertColor = (level: AlertLevel): string => {
  switch (level) {
    case AlertLevel.Warning:
      return yellow;
    case AlertLevel.Info:
      return blue;
    case AlertLevel.Success:
      return green;
    default:
      return red;
  }
};

// change style based on level
const StyledAlert = styled<{ level: AlertLevel }, 'div'>('div')`
  background-color: ${({ level }) => alertColor(level)};
  color: ${white};
  padding: 0.75rem;
  position: relative;
  z-index: 9999;
`;

export const CloseIcon = styled.i`
  cursor: pointer;
`;

export default Alert;
