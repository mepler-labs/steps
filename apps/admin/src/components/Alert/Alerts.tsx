import { Alert as AlertType } from './types';
import * as React from 'react';
import { Alert } from './Alert';
import styled from 'styled-components';

interface Props {
  alerts: AlertType[];
}

export const Alerts = ({ alerts }: Props) => {
  return (
    <StyledAlerts>
      {alerts.map(({ level, message }, i) => {
        return (
          <Alert key={i} level={level}>
            {message}
          </Alert>
        );
      })}
    </StyledAlerts>
  );
};

const StyledAlerts = styled.div``;

export default Alerts;
