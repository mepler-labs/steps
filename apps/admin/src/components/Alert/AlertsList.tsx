import { Alert as AlertType } from './types';
import * as React from 'react';
import { Alert } from './Alert';
import styled from 'styled-components';

interface Props {
  alerts: AlertType[];
  actions: { removeAlert };
}

export const AlertsList = ({ alerts, actions }: Props) => {
  return (
    <StyledAlerts>
      {alerts.map(({ level, message, id }, i) => {
        return (
          <Alert onClose={() => actions.removeAlert(id)} key={i} level={level}>
            {message}
          </Alert>
        );
      })}
    </StyledAlerts>
  );
};

const StyledAlerts = styled.div``;

export default AlertsList;
