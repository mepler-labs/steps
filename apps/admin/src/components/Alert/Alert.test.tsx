import { shallow } from 'enzyme';
import 'jest';
import * as React from 'react';
import { Alert } from './Alert';
import { AlertLevel } from './types';

describe('Alert', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Alert level={AlertLevel.Error}>Some alert message</Alert>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
