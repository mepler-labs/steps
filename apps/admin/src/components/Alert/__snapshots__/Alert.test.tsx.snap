// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<styled.div
  level="error"
>
  <Flex
    alignItems="center"
    justifyContent="space-between"
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    Some alert message
    <styled.i
      className="material-icons"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
    >
      clear
    </styled.i>
  </Flex>
</styled.div>
`;
